yaksh/live_server_tests/selenium_test.py

```diff
@@ -91,72 +156,368 @@ def test_c_question(self, question_label):
+        self.driver.find_element_by_link_text("Back to Courses").click()
+
+    def add_lesson(self):
+        # Open lessons page
+        self.driver.find_element_by_link_text("Add/View Lessons").click()
+
+        # Click add new lesson button
+        self.driver.find_element_by_link_text("Add new Lesson").click()
+
+        # Fill name
+        self.set_name("demo_lesson")
+
+        # Fill description
+        self.set_description("demo_description")
+
+        # Save
+        self.submit()
+
+        # Open all courses page
+        self.driver.find_element_by_link_text("View all Courses").click()
+
+        # Open design learning module page
+        self.driver.find_element_by_partial_link_text(
+            "Add Quizzes/Lessons for").click()
+
+        # Add lesson to module
+        lesson_check_box = WebDriverWait(self.driver, 10).until(
+            EC.presence_of_element_located(
+                (By.XPATH,
+                 "//*[@id='fixed-available']/ul/li[last()]/label/input")
+            )
         )
-        quit_link_elem.click()
+        lesson_check_box.click()
+        self.submit()
+
+        # Get back to courses page
+        self.driver.find_element_by_link_text(
+            "Back to Learning Modules").click()
+
+    def switch_to_student(self):
+        # Switch to student
+        self.driver.find_element_by_id("user_dropdown").click()
+        self.driver.find_element_by_link_text("Switch To Student").click()
+
+    def search_course(self, course_code):
+        # Search the course
+        search_box = WebDriverWait(self.driver, 10).until(
+            EC.presence_of_element_located((By.NAME, "course_code"))
+        )
+        search_box.send_keys(course_code)
+        self.driver.find_element_by_id("id_search").click()
 
-        quit_link_elem = WebDriverWait(self.driver, 5).until(
-            EC.presence_of_element_located((By.NAME, "yes"))
+    def enroll(self):
+        # Click enroll
+        self.driver.find_element_by_link_text("ENROLL").click()
+
+    def switch_to_moderator(self):
+        # Switch to moderator
+        WebDriverWait(self.driver, 10).until(
+            EC.presence_of_element_located((By.ID, "user_dropdown"))
+        ).click()
+        self.driver.find_element_by_link_text("Switch To Moderator").click()
+
+    def open_course_detail(self):
+        WebDriverWait(self.driver, 10).until(
+            EC.presence_of_element_located((By.LINK_TEXT, self.course_name))
+        ).click()
+
+    def check_enroll(self, username):
+        # Check whether the student is enrolled or not
+        WebDriverWait(
+            self.driver, 10).until(
+            EC.presence_of_element_located(
+                (By.XPATH, "//*[@id='enrolled_table']/tbody/tr/td[3]")))
+        soup = BeautifulSoup(self.driver.page_source, features="lxml")
+        cell = soup.find("table").find(
+            "tbody").find_all("tr")[0].find_all("td")[2]
+        return (cell.get_text().lower().strip() == username)
```

Comment:
Do you need beautiful soup for parsing through table? I guess Selenium can do it for you,

Reply:
Done. Removed beautiful soup and now using selenium only to get the text.